src/android/HockeyApp.java

```diff
@@ -27,14 +31,67 @@
 import java.lang.Thread;
 import java.text.SimpleDateFormat;
 import java.util.Calendar;
+import java.util.ArrayList;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.view.View;
 
 public class HockeyApp extends CordovaPlugin {
 
     public static boolean initialized = false;
     public static String appId;
+    public static Object monitor = new Object();
+    public static volatile Bitmap bitmap;
     
     private ConfiguredCrashManagerListener crashListener;
+    
+    // Integration with Crosswalk requires:
+    // - Crosswalk engine version 18 or later
+    // - Latest cordova-plugin-crosswalk-webview plugin     
+    private Bitmap getBitmap() {
+        bitmap = null;
+        boolean isCrosswalk = false;
+        try {
+            Class.forName("org.crosswalk.engine.XWalkWebViewEngine");
+            isCrosswalk = true;
+        } catch (Exception e) {
+        }
+
+        if (isCrosswalk) {
+            long start = System.currentTimeMillis();
+            webView.getPluginManager().postMessage("captureXWalkBitmap", this);
+            try {
+                // Can't take more than 5 seconds to get a screenshot
+                while (bitmap == null && System.currentTimeMillis() - start < 5000) {
+                    synchronized (monitor) {
+                        monitor.wait(50);
```

Comment:
Why don't we simply wait for `5000` millis on the monitor, instead of having the while loop? If the message comes before that, calling `notify()` on the monitor should wake up the thread right? Also, should we extract this `5000` to a `XWALK_SCREENSHOT_WAIT_MILLIS` const at the top so it's easy to tweak in the future?